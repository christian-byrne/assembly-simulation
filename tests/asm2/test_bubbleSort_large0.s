.data

COMMA_SEP:
	.asciiz ", "
    
.globl   printInts
printInts:
	.byte 0

.globl   printWords
printWords:
	.byte 0

.globl   bubbleSort
bubbleSort:
	.byte 1

.globl   printInts_howToFindLen
printInts_howToFindLen:
	.half 0

.globl   intsArray
intsArray:
	.word 10754
	.word 1898
	.word 4196
	.word 11556
	.word 8494
	.word 2042
	.word 13582
	.word 1683
	.word 2998
	.word 9467
	.word 92
	.word 11628
	.word 9937
	.word 3334
	.word 1651
	.word 10053
	.word 14868
	.word 1215
	.word 11628
	.word 12385
	.word 7014
	.word 7897
	.word 5259
	.word 5265
	.word 4394
	.word 1682
	.word 5393
	.word 6876
	.word 10638
	.word 11481
	.word 13728
	.word 4573
	.word 1594
	.word 12881
	.word 6503
	.word 13114
	.word 1009
	.word 8029
	.word 1140
	.word 2244
	.word 4515
	.word 7165
	.word 1008
	.word 10421
	.word 8209
	.word 9664
	.word 9317
	.word 8470
	.word 1849
	.word 13169

.globl   intsArray_END
intsArray_END:
	

.globl   intsArray_len
intsArray_len:
	.word 50

.globl   theString
theString:
	.asciiz "DO NOT TOUCH!"


.globl   PPILnYZuHiPg
PPILnYZuHiPg:
	.asciiz "\n+-------------------------------+\n|    Testcase Variable Dump:    |\n+-------------------------------+\n\n"

.globl   bzFUWntBpjAu
bzFUWntBpjAu:
	.asciiz "==============================================\nTEST (bubbleSort_large0):\n\nprintInts=0\nprintWords=0\nbubbleSort=1\nprintInts_howToFindLen=0\nintsArray=[10754, 1898, 4196, 11556, 8494, 2042, 13582, 1683, 2998, 9467, 92, 11628, 9937, 3334, 1651, 10053, 14868, 1215, 11628, 12385, 7014, 7897, 5259, 5265, 4394, 1682, 5393, 6876, 10638, 11481, 13728, 4573, 1594, 12881, 6503, 13114, 1009, 8029, 1140, 2244, 4515, 7165, 1008, 10421, 8209, 9664, 9317, 8470, 1849, 13169]\nintsArray_len=50\ntheString=\"DO NOT TOUCH!\"\n\n==============================================\n\n"


.text

.globl main

main:
	addi $v0, $zero, 4	# print
	la $a0, bzFUWntBpjAu
	syscall

	jal studentMain

.text
	addi $v0, $zero, 4	# print
	la $a0, PPILnYZuHiPg
	syscall

	addi $v0, $zero, 11	# print
	addi $a0, $zero, '['
	syscall

	la $t0, intsArray
    
	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4
	
	addi $v0, $zero, 4	# print
	la $a0, COMMA_SEP
	syscall

	addi $v0, $zero, 1	# print
	lw $a0, 0($t0)
	syscall
	
	addi $t0, $t0, 4

	add $t0, $zero, $zero   # reset t0
    
	addi $v0, $zero, 11	# print
	addi $a0, $zero, ']'
	syscall
	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall

	addi $v0, $zero, 4	# print
	la $a0, theString
	syscall

	addi $v0, $zero, 11
	addi $a0, $zero, '\n'
	syscall


	addi $v0, $zero, 10
	syscall
