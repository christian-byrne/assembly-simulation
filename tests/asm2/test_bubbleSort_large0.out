==============================================
TEST (bubbleSort_large0):

printInts=0
printWords=0
bubbleSort=1
printInts_howToFindLen=0
intsArray=[10754, 1898, 4196, 11556, 8494, 2042, 13582, 1683, 2998, 9467, 92, 11628, 9937, 3334, 1651, 10053, 14868, 1215, 11628, 12385, 7014, 7897, 5259, 5265, 4394, 1682, 5393, 6876, 10638, 11481, 13728, 4573, 1594, 12881, 6503, 13114, 1009, 8029, 1140, 2244, 4515, 7165, 1008, 10421, 8209, 9664, 9317, 8470, 1849, 13169]
intsArray_len=50
theString="DO NOT TOUCH!"

==============================================

Swap at: 0
Swap at: 1
Swap at: 3
Swap at: 4
Swap at: 6
Swap at: 7
Swap at: 8
Swap at: 9
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 43
Swap at: 44
Swap at: 45
Swap at: 46
Swap at: 47
Swap at: 48
Swap at: 2
Swap at: 3
Swap at: 5
Swap at: 6
Swap at: 7
Swap at: 8
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 43
Swap at: 44
Swap at: 45
Swap at: 46
Swap at: 47
Swap at: 2
Swap at: 4
Swap at: 5
Swap at: 6
Swap at: 7
Swap at: 9
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 14
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 43
Swap at: 44
Swap at: 45
Swap at: 46
Swap at: 1
Swap at: 3
Swap at: 4
Swap at: 6
Swap at: 8
Swap at: 9
Swap at: 10
Swap at: 11
Swap at: 13
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 27
Swap at: 28
Swap at: 30
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 43
Swap at: 44
Swap at: 2
Swap at: 3
Swap at: 5
Swap at: 8
Swap at: 9
Swap at: 12
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 26
Swap at: 27
Swap at: 29
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 43
Swap at: 1
Swap at: 4
Swap at: 7
Swap at: 8
Swap at: 11
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 25
Swap at: 26
Swap at: 28
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 42
Swap at: 0
Swap at: 3
Swap at: 6
Swap at: 7
Swap at: 10
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 24
Swap at: 25
Swap at: 27
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 41
Swap at: 2
Swap at: 5
Swap at: 6
Swap at: 9
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 23
Swap at: 24
Swap at: 26
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 40
Swap at: 1
Swap at: 5
Swap at: 8
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 22
Swap at: 23
Swap at: 25
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 39
Swap at: 0
Swap at: 4
Swap at: 7
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 21
Swap at: 22
Swap at: 24
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 38
Swap at: 3
Swap at: 6
Swap at: 9
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 20
Swap at: 21
Swap at: 23
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 37
Swap at: 2
Swap at: 5
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 19
Swap at: 20
Swap at: 22
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 36
Swap at: 1
Swap at: 4
Swap at: 9
Swap at: 10
Swap at: 11
Swap at: 12
Swap at: 13
Swap at: 14
Swap at: 18
Swap at: 19
Swap at: 21
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 29
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 35
Swap at: 3
Swap at: 10
Swap at: 11
Swap at: 17
Swap at: 18
Swap at: 20
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 28
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 34
Swap at: 2
Swap at: 9
Swap at: 10
Swap at: 16
Swap at: 17
Swap at: 19
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 27
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 33
Swap at: 1
Swap at: 9
Swap at: 15
Swap at: 16
Swap at: 18
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 26
Swap at: 28
Swap at: 30
Swap at: 31
Swap at: 32
Swap at: 8
Swap at: 14
Swap at: 15
Swap at: 17
Swap at: 19
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 25
Swap at: 27
Swap at: 29
Swap at: 30
Swap at: 31
Swap at: 7
Swap at: 13
Swap at: 14
Swap at: 16
Swap at: 18
Swap at: 20
Swap at: 21
Swap at: 22
Swap at: 23
Swap at: 24
Swap at: 29
Swap at: 30
Swap at: 6
Swap at: 12
Swap at: 13
Swap at: 17
Swap at: 19
Swap at: 20
Swap at: 21
Swap at: 23
Swap at: 28
Swap at: 29
Swap at: 5
Swap at: 11
Swap at: 12
Swap at: 16
Swap at: 18
Swap at: 19
Swap at: 20
Swap at: 22
Swap at: 28
Swap at: 4
Swap at: 11
Swap at: 15
Swap at: 17
Swap at: 18
Swap at: 19
Swap at: 21
Swap at: 27
Swap at: 3
Swap at: 10
Swap at: 14
Swap at: 16
Swap at: 17
Swap at: 18
Swap at: 20
Swap at: 26
Swap at: 9
Swap at: 13
Swap at: 15
Swap at: 16
Swap at: 17
Swap at: 19
Swap at: 25
Swap at: 8
Swap at: 12
Swap at: 14
Swap at: 15
Swap at: 16
Swap at: 18
Swap at: 24
Swap at: 7
Swap at: 11
Swap at: 13
Swap at: 14
Swap at: 15
Swap at: 17
Swap at: 23
Swap at: 6
Swap at: 10
Swap at: 12
Swap at: 13
Swap at: 16
Swap at: 22
Swap at: 5
Swap at: 9
Swap at: 11
Swap at: 12
Swap at: 15
Swap at: 21
Swap at: 4
Swap at: 8
Swap at: 10
Swap at: 11
Swap at: 14
Swap at: 20
Swap at: 3
Swap at: 7
Swap at: 9
Swap at: 10
Swap at: 13
Swap at: 19
Swap at: 2
Swap at: 6
Swap at: 8
Swap at: 12
Swap at: 18
Swap at: 5
Swap at: 7
Swap at: 11
Swap at: 17
Swap at: 4
Swap at: 6
Swap at: 10
Swap at: 16
Swap at: 3
Swap at: 5
Swap at: 9
Swap at: 15
Swap at: 2
Swap at: 4
Swap at: 8
Swap at: 14
Swap at: 1
Swap at: 3
Swap at: 7
Swap at: 13
Swap at: 2
Swap at: 6
Swap at: 12
Swap at: 5
Swap at: 11
Swap at: 4
Swap at: 10
Swap at: 3
Swap at: 9
Swap at: 2
Swap at: 1

+-------------------------------+
|    Testcase Variable Dump:    |
+-------------------------------+

[92, 1008, 1009, 1140, 1215, 1594, 1651, 1682, 1683, 1849, 1898, 2042, 2244, 2998, 3334, 4196, 4394, 4515, 4573, 5259, 5265, 5393, 6503, 6876, 7014, 7165, 7897, 8029, 8209, 8470, 8494, 9317, 9467, 9664, 9937, 10053, 10421, 10638, 10754, 11481, 11556, 11628, 11628, 12385, 12881, 13114, 13169, 13582, 13728, 14868]
DO NOT TOUCH!
